alanlawrance/steam-deploy-public
# Security layer

Hello there,

in this pull request I integrated the HMAC SH256 signature check sent by Unity Cloud Build, doing so we can prevent not authorized post request from pushing into our depots.

This branch is based on the automatic DRM branch due to the huge refactoring done there.

By default the signature check is disabled, only once someone add a .env file with the required key then it will get checked (there is a sample example.env there, it need to be renamed ".env"). Then on Unity Cloud Build integration page you'll need to add an authorization secret of type HMAC-SHA256 Signature.

How to test:
1. Switch to that branch, pull and do npm install and restart the app
2. Make a build without changing anything, should build properly
3. Add the security and make the build, should build properly

What is missing:
- Tutorial on how to use

Thanks,
Paolo
